src/wp-includes/media.php

```diff
@@ -5455,6 +5459,21 @@ function wp_get_loading_attr_default( $context ) {
 		return false;
 	}
 
+	// Conditionally skip lazy-loading on images before the loop.
+	if ( 'wp_get_attachment_image' === $context || 'the_post_thumbnail' === $context ) {
+		// Only apply for main query but before the loop.
+		if ( $wp_query->before_loop && is_main_query() ) {
```

Comment:
```suggestion
		if ( $wp_query->before_loop && $wp_query->is_main_query() ) {
```

Reply:
What if `wp_get_loading_attr_default` is called outside the context of WP_Query? Like site icon or similar.

Reply:
Wherever `wp_get_loading_attr_default()` is called, it depends on the context on what it should do. I'm not sure what you mean in regards to site icon, since I don't think that relies on the function anywhere.

The current logic here is in fact _especially_ for when logic is called outside of `WP_Query`, namely if it is called before the query loop is started. In that case, we should avoid lazy-loading as the image is likely in the header of the page.

Reply:
Site icon is example of where an image is output the context of a WP_Query. Just trying to understand what happens if call output the loop.

Reply:
@spacedmonkey My point is that the site icon is never output as an `img` tag, at least there is no WP core function that does that. You would need to use `wp_get_attachment_image()` with the site icon ID to do that, at which point it uses the `wp_get_attachment_image` context anyway.

Reply:
@spacedmonkey Are you referring to the Site Logo?

I noticed that [the Site Logo Block](https://github.com/WordPress/gutenberg/blob/37dc9f67575153857b59550fc964cc2954f45ce2/packages/block-library/src/site-logo/index.php#L26) calls `get_custom_logo()`, which calls `wp_get_attachment_image()` [here](https://github.com/WordPress/wordpress-develop/blob/trunk/src/wp-includes/general-template.php#L1099) and outputs the Site Logo as an `img` tag.

From what I can tell, there are no changes to context during this so it should use the `wp_get_attachment_image` context as @felixarntz mentions.

Reply:
@spacedmonkey 
> I am asking about calls to `wp_get_attachment_image` outside of WP_Query.

That's precisely what this change is for: If e.g. a page template or template part calls `wp_get_attachment_image()` (outside of WP_Query), today all of these images would get lazy-loaded by default, which is problematic e.g. for images in the header. This change ensures that `wp_get_attachment_image()` images that e.g. appear in the `header.php` template part are not lazy-loaded.